src/main/java/org/folio/linked/data/service/resource/meta/MetadataServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package org.folio.linked.data.service.resource.meta;
+
+import static org.folio.ld.dictionary.ResourceTypeDictionary.INSTANCE;
+import static org.folio.linked.data.model.entity.ResourceSource.LINKED_DATA;
+
+import java.util.UUID;
+import java.util.function.Consumer;
+import org.folio.linked.data.model.entity.InstanceMetadata;
+import org.folio.linked.data.model.entity.Resource;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MetadataServiceImpl implements MetadataService {
+
+  @Override
+  public void ensureMetadata(Resource resource) {
+    ensure(resource, metadata ->
+      metadata.setInventoryId(UUID.randomUUID().toString())
+    );
+  }
+
+  @Override
+  public void ensureMetadata(Resource resource, InstanceMetadata oldResourceMeta) {
```

Comment:
minor
just `ensure`
It's rule not use the same naming in method as a main class
example
UserService -> getById()
not
UsereService -> getUserById

Reply:
makes sense, done